mitodl/open-discussions
# For field channel API serializer, return "lists" attribute as ordered list of UserLists

#### Pre-Flight checklist


#### What are the relevant tickets?
Related to #3642

#### What's this PR do?
Refactors the `FieldChannelSerializer` to return `lists` as an ordered list of `UserListSerializer` items.

#### How should this be manually tested?
- Go to `http://localhost:8063/api/v0/fields/` while logged in as an admin user, if there are already some existing field channels then you should see them listed, and the `lists` attribute if any should be as expected.
- Create several new public user lists from `http://localhost:8063/learn` (click the bookmark icon for any ocw course, click 'Create New List', fill out the form using 'Public' for privacy, and save).  Create a private list too.
- Create a new `FieldChannel` by posting the following JSON, you should get a valid response.
    ```
    {
        "name": "new_field_2",
        "title": "New Field 2",
        "description": "New Field 2",
        "lists": [<public_list_1_id>, <public_list_2_id>],
        "featured_list": <public_list_1_id>
    }
   ```

- Go to `http://localhost:8063/api/v0/fields/new_field_2`, the details on your new field should be listed, including lists and subfields.

- PATCH the following JSON to update the data including lists and subfields:
    ```
    {
        "title": "New Field 2 modified",
        "description": "New Field 2 new desc",
        "lists": [<public_list_2_id>],
        "featured_list": <public_list_2_id>,
    }
    ```
